homework/vector-of-shared-ptrs/vectorFunctions.cpp

```diff
@@ -0,0 +1,39 @@
+#include "vectorFunctions.hpp"
+#include <algorithm>
+#include <iostream>
+
+std::vector<std::shared_ptr<int>> generate(int countNumber) {
+    std::vector<std::shared_ptr<int>> result(countNumber);
+    std::for_each(begin(result), end(result),
+                  [number = 0](auto& pointer) mutable {
+                      pointer = std::make_shared<int>(number++);
+                  });
```

Comment:
use std::generate. Don't use for_each when other algorithm fit better. Also remember raw loop is also good.